File: Sources/Research Projects/Gesture Research-OC/Gesture Research-OC/ViewController.m
//
//  ViewController.m
//  Gesture Research-OC
//
//  Created by CL Wang on 2/6/23.
//

#import "ViewController.h"

@interface ViewController()

@property (strong) NSMutableDictionary *twoFingersTouches;

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];

//    self.view.allowedTouchTypes = NSTouchTypeMaskDirect | NSTouchTypeMaskIndirect;
//    [NSTouch.TouchTypeMask.direct, NSTouch.TouchTypeMask.indirect];
    // Do any additional setup after loading the view.
}


- (void)setRepresentedObject:(id)representedObject {
    [super setRepresentedObject:representedObject];

    // Update the view, if already loaded.
}


@end

File: Sources/Research Projects/Gesture Research/Gesture Research/ViewController.swift
//
//  ViewController.swift
//  Gesture Research
//
//  Created by CL Wang on 2/6/23.
//

import Cocoa

class ClipView: NSClipView {
    
    override var isFlipped: Bool { true }
    
}

class ViewController: NSViewController {

    @IBOutlet weak var scrollView: NSScrollView!
    
    override func viewDidLoad() {
        super.viewDidLoad()

        let view = NSView(frame: NSMakeRect(0, 0, 2000, 1000))
        view.wantsLayer = true
        view.layer!.backgroundColor = NSColor.systemGreen.cgColor
        
        scrollView.contentView = ClipView()
        scrollView.documentView = view
        
        // Do any additional setup after loading the view.
    }

    override var representedObject: Any? {
        didSet {
        // Update the view, if already loaded.
        }
    }
    
    @IBAction func buttonDidClick(_ sender: Any) {
        scrollView.scroll(scrollView.contentView, to: NSMakePoint(200, 0))
    }
    

}

